packages/esm-ward-app/src/hooks/usePatientPendingOrders.ts

```diff
@@ -0,0 +1,95 @@
+import { useEffect, useState } from 'react';
+import { type Concept, openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+import { type Order } from '@openmrs/esm-patient-common-lib';
+import { type Encounter } from '../types';
+
+const encounterRepresentation =
+  'custom:(uuid,encounterDatetime,encounterType,location:(uuid,name),' +
+  'patient:(uuid,display),encounterProviders:(uuid,provider:(uuid,name)),' +
+  'obs:(uuid,obsDatetime,voided,groupMembers,formFieldNamespace,formFieldPath,order:(uuid,display),concept:(uuid,name:(uuid,name)),' +
+  'value:(uuid,display,name:(uuid,name),names:(uuid,conceptNameType,name))))';
+
+const conceptRepresentation =
+  'custom:(uuid,display,name,datatype,set,answers,hiNormal,hiAbsolute,hiCritical,lowNormal,lowAbsolute,lowCritical,units,' +
+  'setMembers:(uuid,display,answers,datatype,hiNormal,hiAbsolute,hiCritical,lowNormal,lowAbsolute,lowCritical,units))';
+
+/**
+ * Custom hook to count orders of type TestOrder that lack associated observations.
+ *
+ * This hook takes an array of orders and counts how many of them do not have observations recorded.
+ * It fetches the necessary concept and encounter data for each order to determine if observations are missing.
+ *
+ * @param {Order[]} orders - An array of order objects. Each order contains information about the order, including concept and encounter UUIDs.
+ * @returns {count: number, isLoading: boolean } - The count of orders without observations, and a loading state.
+ *
+ * @example
+ * const orders = [
+ *   { concept: { uuid: 'concept-uuid-1' }, encounter: { uuid: 'encounter-uuid-1' }, ... },
+ *   { concept: { uuid: 'concept-uuid-2' }, encounter: { uuid: 'encounter-uuid-2' }, ... },
+ * ];
+ *
+ * const {count: unobservedCount, isLoading } = useCountTestOrdersWithoutObs(testOrders);
+ * // unobservedCount would be the number of test orders without associated observations.
+ */
+
+export function useCountTestOrdersWithoutObs(orders: Order[]): { count: number; isLoading: boolean } {
+  const [countPending, setCountPending] = useState(0);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchOrderData = async (order: Order) => {
+      const conceptKey = `${restBaseUrl}/concept/${order.concept.uuid}?v=${conceptRepresentation}`;
+      const encounterKey = `${restBaseUrl}/encounter/${order.encounter.uuid}?v=${encounterRepresentation}`;
+
+      const [concept, encounter] = await Promise.all([
+        openmrsFetch<Concept>(conceptKey),
+        openmrsFetch<Encounter>(encounterKey),
+      ]);
+
+      return { concept, encounter };
+    };
+
+    const countPendingOrders = async () => {
+      setIsLoading(true);
+      try {
+        const pendingCounts = await Promise.all(
+          orders.map(async (order) => {
+            if (order.orderType['javaClassName'] === 'org.openmrs.TestOrder') {
+              const { concept, encounter } = await fetchOrderData(order);
```

Comment:
Is the assumption here that the test result obs is going to be on the same encounter as the order?  I don't think that will normally be the case, unfortunately.

Reply:
It also looked like according to the ticket this should be limited to orders placed on the current visit (which should hopefully simplify things)

Reply:
@usamaidrsk please answer Mark's first question

I would be inclined to have the caller of this hook handle filtering orders down to only the ones placed on this visit. The filtered list of orders would be passed in as the argument. So here the change would go in ward-patient-pending-orders. @usamaidrsk can you make that change?

Reply:
> Is the assumption here that the test result obs is going to be on the same encounter as the order?

Looking at a similar implementation from the [here](https://github.com/openmrs/openmrs-esm-patient-chart/blob/59934a3330a02eb0e41e79df4cabedacdda77f52/packages/esm-patient-orders-app/src/components/test-order.component.tsx#L28), looks like that is the case.

Reply:
> I would be inclined to have the caller of this hook handle filtering orders down to only the ones placed on this visit. The filtered list of orders would be passed in as the argument. So here the change would go in ward-patient-pending-orders. @usamaidrsk can you make that change?

Will do @brandones. Thanks

Reply:
> It also looked like according to the ticket this should be limited to orders placed on the current visit (which should hopefully simplify things)

Is it possible querying orders by a specific visit encounter? cc @mogoodrich @brandones 

Note: to fetch orders I used the [usePatientOrders](https://github.com/openmrs/openmrs-esm-patient-chart/blob/59934a3330a02eb0e41e79df4cabedacdda77f52/packages/esm-patient-common-lib/src/orders/useOrders.ts#L16) from the `esm-patient-common-lib`

Reply:
Looks like you might be correct about the test orders, let me start a discussion on Slack (or elsewhere)

Reply:
The discussion on Slack: https://openmrs.slack.com/archives/CHP5QAE5R/p1721851387699259

Reply:
@mogoodrich , do you have suggestions about implementation? Based on the Slack thread, it sounds like we may need to query obs by `order_id`?

Reply:
@mseaton  had suggested on the ticket (https://openmrs.atlassian.net/browse/O3-3223) moving this to the backend, which offhand made sense to me, I was planning on following up with him today and then one of us can update the ticket/PR.